src/mono/mono/mini/mini.h

```diff
@@ -2970,6 +2966,26 @@ mini_safepoints_enabled (void)
 #endif
 }
 
+static inline gboolean
+mini_class_is_simd (MonoCompile *cfg, MonoClass *klass)
+{
+#ifdef MONO_ARCH_SIMD_INTRINSICS
+	if (!(((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass)))
+		return FALSE;
+	if (COMPILE_LLVM (cfg))
+		return TRUE;
+	int size = mono_type_size (m_class_get_byval_arg (klass), NULL);
+#ifdef TARGET_ARM64
+	if (size == 8 || size == 16)
+		return TRUE;
+#else
+	if (size == 16)
+		return TRUE;
```

Comment:
We emit some intrinsits for Vector. If I remember correctly, Vector could be either 16 bytes or 32 bytes on Amd64.

Reply:
This is the same logic we had previously.